### ISSUE TITLE
Extend operands_and_partials to work with multi-variate containers

### ISSUE THREAD

**Issue by wds15** [stan-dev/math#707]
#### Summary:
The `operands_and_partials` facility has almost complete support for multi-variate containers (arrays of vectors and plain vectors as input). 

#### Description:
The `partials_vec_` member is only defined whenever a vector of `vars` is passed in as an edge, but when the edge is to be ignored, then there is no definition of any `partials_vec_` which prevents code to compile.

#### Reproducible Steps:
So this works:

```C++
  operands_and_partials<std::vector<vector_v>, std::vector<vector_d>, vector_v>
      o4(v_vec, d_vec_vec, v_vec2);
  o4.edge1_.partials_vec_[0] += d_vec1;
  o4.edge3_.partials_vec_[0] += d_vec2;
```

but if edge3 is passed in as `vector_d`, then there is no `partials_vec_` such that code will not compile. 

#### Current Output:
This functionality is not yet used, but I want to submit a `multi_normal_cholesky` which brings analytic gradients.

#### Expected Output:
It should just work which means in this case to compile.

#### Additional Information:
Provide any additional information here.

#### Current Version:
v2.17.0

**Comment by wds15** [comment]
The idea to introduce a `empty_nested_broadcast_array` seems to solve all issues we have. Now the sizeof operator goes from 5 to 10 (instead to 80 when using broadcast_array on the empty_broadcast_array).

I don't think we can do much better than that, but others (in particular @seantalts ) should check.

**Comment by wds15** [comment]
This has been fixed by pull #711
